tests/e2e/system_test/test_object_expiration.py

```diff
@@ -1,96 +1,413 @@
-                "ID": "data-expire",
-                "Status": "Enabled",
-            }
-        ]
-    }
+class TestObjectExpiration:
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    @system_test
+    @bugzilla("2039309")
+    @skipif_ocs_version("<4.11")
+    @pytest.mark.polarion_id("OCS-4852")
+    @magenta_squad
+    def test_object_expiration(self, mcg_obj, bucket_factory):
+        """
+        Test object expiration, see if the object is deleted within the expiration + 8 hours buffer time
+
+        """
+        # Creating S3 bucket
+        bucket = bucket_factory()[0].name
+        object_key = "ObjKey-" + str(uuid.uuid4().hex)
+        obj_data = "Random data" + str(uuid.uuid4().hex)
+        expiration_days = 1
+        buffer_time_in_hours = 8
+
+        expire_rule_4_10 = {
+            "Rules": [
+                {
+                    "Expiration": {
+                        "Days": expiration_days,
+                        "ExpiredObjectDeleteMarker": False,
+                    },
+                    "ID": "data-expire",
+                    "Prefix": "",
+                    "Status": "Enabled",
+                }
+            ]
+        }
+        expire_rule = {
+            "Rules": [
+                {
+                    "Expiration": {
+                        "Days": expiration_days,
+                        "ExpiredObjectDeleteMarker": False,
+                    },
+                    "Filter": {"Prefix": ""},
+                    "ID": "data-expire",
+                    "Status": "Enabled",
+                }
+            ]
+        }
+
+        logger.info(f"Setting object expiration on bucket: {bucket}")
+        if version.get_semantic_ocs_version_from_config() < version.VERSION_4_11:
+            mcg_obj.s3_client.put_bucket_lifecycle_configuration(
+                Bucket=bucket, LifecycleConfiguration=expire_rule_4_10
+            )
+        else:
+            mcg_obj.s3_client.put_bucket_lifecycle_configuration(
+                Bucket=bucket, LifecycleConfiguration=expire_rule
+            )
+
+        logger.info(f"Getting object expiration configuration from bucket: {bucket}")
+        logger.info(
+            f"Got configuration: {mcg_obj.s3_client.get_bucket_lifecycle_configuration(Bucket=bucket)}"
+        )
+
+        logger.info(f"Writing {object_key} to bucket: {bucket}")
+        assert s3_put_object(
+            s3_obj=mcg_obj, bucketname=bucket, object_key=object_key, data=obj_data
+        ), "Failed: Put Object"
 
-    logger.info(f"Setting object expiration on bucket: {bucket}")
-    if version.get_semantic_ocs_version_from_config() < version.VERSION_4_11:
-        mcg_obj.s3_client.put_bucket_lifecycle_configuration(
-            Bucket=bucket, LifecycleConfiguration=expire_rule_4_10
+        logger.info("Waiting for 1 day + 8 hours buffer time")
```

Comment:
Are we running the script by tweaking the wait time, or should we indeed wait for 1 day plus buffer time?

Reply:
yeah i missed to change this.